disnake/ext/commands/interaction_bot_base.py

```diff
@@ -698,65 +767,75 @@ async def _sync_application_commands(self) -> None:
         if not isinstance(self, disnake.Client):
             raise NotImplementedError("This method is only usable in disnake.Client subclasses")
 
-        if not self._sync_commands or self._is_closed or self.loop.is_closed():
+        if not self._command_sync._sync_enabled or self._is_closed or self.loop.is_closed():
             return
 
         # We assume that all commands are already cached.
         # Sort all invokable commands between guild IDs:
         global_cmds, guild_cmds = self._ordered_unsynced_commands(self._test_guilds)
-        if global_cmds is None:
-            return
 
-        # Update global commands first
-        diff = _app_commands_diff(
-            global_cmds, self._connection._global_application_commands.values()
-        )
-        update_required = bool(diff["upsert"]) or bool(diff["edit"]) or bool(diff["delete"])
-
-        # Show the difference
-        self._log_sync_debug(
-            "Application command synchronization:\n"
-            "GLOBAL COMMANDS\n"
-            "===============\n"
-            f"| Update is required: {update_required}\n{_format_diff(diff)}"
-        )
+        if self._command_sync.sync_global_commands and global_cmds is not None:
+            # Update global commands first
+            diff = _app_commands_diff(
+                global_cmds, self._connection._global_application_commands.values()
+            )
+            if not self._command_sync.allow_command_deletion:
+                # because allow_command_deletion is disabled, we want to never delete a command, so we move the delete commands to no_changes
```

Comment:
"<...> we move the delete commands to delete_ignored"

Reply:
https://github.com/DisnakeDev/disnake/pull/806/commits/34095b50f7b7021e1f18ddc55f60ebde30c1ed4b